File: SurveyPoint35/QMS/QMSWeb/includes/clsLog.vb
Imports System.Text
Imports log4net

Public Class clsLog
    Public Shared Sub LogError(ByVal type As System.Type, ByVal msg As String, ByVal ex As Exception)
        Dim sb As StringBuilder = New StringBuilder

        sb.AppendFormat("{0}: ", msg)
        sb.AppendFormat("{0}, ", HttpContext.Current.Request.RawUrl())
        If IsNumeric(HttpContext.Current.User.Identity.Name) Then
            sb.AppendFormat("User {0}, ", HttpContext.Current.User.Identity.Name)
        End If
        Dim log As ILog = LogManager.GetLogger(type)
        log.Error(sb.ToString(), ex)
    End Sub
End Class

File: SurveyPoint35/QMS/CommonTools/clsLog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CommonTools
{
    /// <summary>
    /// Summary description for clsLog.
    /// </summary>
    public class clsLog
    {
        public clsLog()
        {

        }

        public static void LogWarn(System.Type type, String msg, Exception ex)
        {
            //TODO:  IMplement
            //ILog log = LogManager.GetLogger(type);
            //if (log.IsWarnEnabled) log.Warn(msg, ex);
        }

        public static void LogError(System.Type type, String msg, Exception ex)
        {
            //TODO:  IMplement
            //ILog log = LogManager.GetLogger(type);
            //if (log.IsErrorEnabled) log.Error(msg, ex);
        }

    }
}